### DISCUSSION TITLE
Embedded iFrame figures

### DISCUSSION THREAD

**Discussion by brutaldigital** [thegetty/quire#660] [Get Help] [answered]
Hi there,

I'm wondering if anyone can advise on the best way to include external resources as a Figure. We'd like to be able to include resources from sources such as SketchFab or the Internet Archive. These would usually be embedded as an iFrame.

Sample resources:

- https://sketchfab.com/3d-models/robert-elizabeth-barrett-browning-hands-ea6372aca91d493ea0e1c61b4b09b2bc
- https://archive.org/stream/agnews-1891-1898?ui=embed#page/n169/mode/2up

Might a workaround to including resources such as this be to follow the guidance for including a table?:

- https://quire.getty.edu/docs-v1/figure-images/#add-tables

**Comment by geealbers** [comment, marked as answer]
@brutaldigital you're right that using the existing system for including a table would be the way to go. In Quire, tables are held as individual HTML files and then added into the project with the figure shortcode. But Quire doesn't really know what HTML is in the file, and so it doesn't need to be a `<table>` but could instead be and `<iframe>` embed or anything else. So, if you save the embeds from those two example, you list them in the figures.yaml as you would normally, but with `media_type: table` and then insert them where you need them. I did a quick test and it worked pretty well, though you're probably going to want to add some CSS to optimize their display.

You also may want a fallback for print. Quire will include anything with `media_type: table` in the PDF output by default, but I think maybe you could add markup for a static image in the HTML file alongside the embed. And then add some media query CSS to show the embed on screen and the fallback in print.

**Reply by brutaldigital** [reply]
Thanks for confirming!
